src/routes/subscriptions.js

```diff
@@ -0,0 +1,79 @@
+const { StatusCodes } = require('http-status-codes');
+
+const express = require('express');
+const router = express.Router();
+const db = require('../storage/DataAccess');
+const COLLECTION = 'plandefs';
+
+// POST /
+// Recieve subscription notification
+router.post('/:id', (req, res) => {
+  const id = req.params.id;
+  const planDef = getPlanDef(id.res);
+  if (planDef) {
+    res.sendStatus(StatusCodes.OK);
+    usePlanDef(planDef);
+  }
+});
+
+// if the subscription returns a resource, the notification
+// will be a PUT instead of a POST
+router.put('/:id/:resource/:resourceId', (req, res) => {
+  const id = req.params.id;
+  const planDef = getPlanDef(id, res);
+  if (planDef) {
+    res.sendStatus(StatusCodes.OK);
+    usePlanDef(planDef, req.body);
+  }
```

Comment:
Since we check `planDef` here I think the code would be cleaner to have `getPlanDef` return `null` and make the calling function send the 404